tests/queries/0_stateless/02157_url_auth.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+# Tags: no-fasttest
+# Tag no-fasttest: Not sure why fail even in sequential mode. Disabled for now to make some progress.
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+# We should have correct env vars from shell_config.sh to run this test
+
+GO_BUILD="${CURDIR}/go_build"
+DOCKER_NAME="url_basic_auth_go_webserver"
+mkdir -p ${GO_BUILD}
+cd ${GO_BUILD}
+cp ../02157_url_basic_auth.go app.go
+cat <<EOF >> Dockerfile
+FROM golang:alpine as builder
+
+RUN apk --no-cache add git
+
+WORKDIR /go/src/webserver
+
+COPY app.go .
+
+RUN go mod init
+
+RUN go mod tidy
+
+RUN CGO_ENABLED=0 go build -a -installsuffix cgo -o app .
+
+FROM alpine:latest as prod
+
+RUN apk --no-cache add ca-certificates
+
+WORKDIR /root/
+
+COPY --from=0 /go/src/webserver/app .
+
+CMD ["./app"]
+EOF
+
+docker build -t go/webserver:1 . >/dev/null
```

Comment:
Test, that builds docker and go application is something new and pretty heavy, I guess it can be converted to test in python?

Reply:
I observed that there's actually one single test in Go for ClickHouse currently though it doesn't vendor anything external: https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/0_stateless/02013_zlib_read_after_eof.go

Reply:
That test exists in go because implementation if zlib in go allows to reproduce the problem (at least it does not build any docker images, though building go app is pretty heavy anyway), but in this case it looks like any http server can serve the purpose

Reply:
Yes true.

Reply:
> Test, that builds docker and go application is something new and pretty heavy, I guess it can be converted to test in python?

I use multi-stage builds to run the go app, which remove dependencies and make the image smaller. I may understand why you think go is heavy because it needs compile. But there's no dependency to conflict the environment after two stage builds in docker. And I've checked it is good. BTW, we use docker in our unit test quit often.

Reply:
I still think that python is enough here. Here you can find simple http server with basic auth example in python - https://gist.github.com/dragermrb/108158f5a284b5fba806

Reply:
I have replaced go test with python. Pls check!